components/optimizer/dummy.py

```diff
@@ -0,0 +1,75 @@
+from components.optimizer.base import BaseOptimizer
+from common.minimodel.plan import Plans, Plan
+from common.minimodel.program import ProgramID
+from common.minimodel.group import Group
+from common.calculations.programinfo import ProgramInfo
+from datetime import datetime, timedelta
+from typing import Mapping
+import random
+
+
+class DummyOptimizer(BaseOptimizer):
+
+    def __init__(self, seed=42):
+        # Set seed for replication
+        random.seed(seed)
+        self.groups = []
+
+    def _allocate_time(self, plan: Plan, obs_time: timedelta) -> datetime:
+        """
+        Allocate time for an observation inside a Plan
+        This should be handle by the optimizer as can vary from algorithm to algorithm
+        """
+        # Get first available slot
+        start = plan.start
+        for v in plan.visits:
+            delta = v.start_time - start + obs_time
+            start += delta
+        return start
+
+    def _run(self, plans: Plans):
+        """
+        Gives a random group/observation to add to plan
+        """
+        
+        while not plans.all_done() and len(self.groups) > 0:
+
+            ran_group = random.choice(self.groups)
+            if self.add(ran_group, plans):
+                # TODO: All observations in the group are being inserted so the whole group
+                # can be removed
+                self.groups.remove(ran_group)
+            else:
+                print('group not added')
+        
+    def setup(self, programInfo: Mapping[ProgramID, ProgramInfo]):
+        """
+        Preparation for the optimizer i.e create chromosomes, etc.
+        """
+        self.groups = []
+        for p in programInfo.values():
+            self.groups.extend([g for g in p.group_data.values() if g.group.is_observation_group()])
+        return self
+    
+    def add(self, group: Group, plans: Plans) -> bool:
+        """
+        Add a group to a Plan
+        This is called when a new group is added to the program
+        """                        
+        # TODO: Missing different logic for different AND/OR GROUPS
+        # Add method should handle those
+        for observation in group.group.observations():
```

Comment:
This has me confused... If `group` is a `Group` as per the method type hints, it shouldn't have a member named `group`?